util/dnsname/dnsname.go

```diff
@@ -17,3 +96,21 @@ func HasSuffix(name, suffix string) bool {
 	nameBase := strings.TrimSuffix(name, suffix)
 	return len(nameBase) < len(name) && strings.HasSuffix(nameBase, ".")
 }
+
+// SanitizeHostname returns hostname with some common suffixes removed.
+func SanitizeHostname(hostname string) string {
+	hostname = strings.TrimSuffix(hostname, ".local")
+	hostname = strings.TrimSuffix(hostname, ".localdomain")
+	hostname = strings.TrimSuffix(hostname, ".lan")
+	return SanitizeLabel(hostname)
+}
+
+// ToMagicDNSBaseName trims any trailing dots from a DNS name and removes
+// the magicDNSSuffix ending if present.
+func ToMagicDNSBaseName(name string, magicDNSSuffix string) string {
```

Comment:
It really has nothing to do with MagicDNS though, right?

I'd just name this TrimSuffix, matching the naming of https://golang.org/pkg/strings/#TrimSuffix (especially as we already have `HasSuffix`).

Then document+test whether this can ever return trailing dots (I think it can currently, which is inconsistent in that it does trim the dots in the case where `!HasSuffix`), as line 111 does it and that's then returned.

Maybe move line 111 down?

Reply:
About the trailing dot - are you saying it should not return a trailing dot in the `!HasSuffix` case? Currently it does remove the trailing dots.